Modules/Course/classes/class.ilObjCourse.php

```diff
@@ -593,16 +593,21 @@ public static function _registrationEnabled(int $a_obj_id) : bool
         return ilObjCourseAccess::_registrationEnabled($a_obj_id);
     }
 
-    public function read()
+    public function read() : void
     {
         parent::read();
         $this->setOrderType(ilContainerSortingSettings::_lookupSortMode($this->getId()));
         $this->__readSettings();
     }
 
-    public function create($a_upload = false)
+    public function create($a_upload = false) : int
```

Comment:
stumbled upon another one:  argument $a_upload without type (bool) in function class.ilObjCourse::create()..

Reply:
Hi @utesche ,
the parent classes of ilObjCourse doesn't use this parameter. So this is only used by ilObjCourse and doesn't fit into this PR.
But thanks again for the detailed look.